internal/prometheusbpint/modules.go

```diff
@@ -0,0 +1,31 @@
+package prometheusbpint
+
+import (
+	"runtime/debug"
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var goModules = promauto.With(GlobalRegistry).NewGaugeVec(prometheus.GaugeOpts{
+	Name: "baseplate_go_modules",
+	Help: "Export the version information for included Go modules, and whether the module is the 'main' module or a 'dependency'.  Always 1",
+}, []string{"go_module", "module_role", "module_replaced", "module_version"})
+
+func RecordModuleVersions(info *debug.BuildInfo) {
```

Comment:
🔕 also, do we want to consider moving this to `prometheusbp` so some more customized services (that don't use `baseplate.New`) can also use it?

Reply:
+1 I think it would be useful to not lock this behind `internal`